osf/management/commands/backfill_domain_references.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+import re
+import logging
+import datetime
+import operator
+from functools import reduce
+
+from django.db.models import Q
+from django.apps import apps
+from django.core.management.base import BaseCommand
+from osf.external.spam.tasks import check_resource_for_domains
+from addons.wiki.models import WikiVersion
+
+
+logger = logging.getLogger(__name__)
+
+DOMAIN_MATCH_REGEX = re.compile(r'(?P<protocol>\w+://)?(?P<www>www\.)?(?P<domain>[\w-]+\.\w+)(?P<path>/\w*)?')
+DOMAIN_SEARCH_REGEX = r'(http://[^ \'}\[\]\~\(\)\/]+|https://[^ \'}\[\]\~\(\)\/]+)'
+
+
+def spawn_tasks_for_domain_references_backfill(model, query, get_guid, batch_size=None, dry_run=False):
+    for item in model.objects.filter(query)[:batch_size]:
+        spam_content = item._get_spam_content()
+        if not dry_run:
+            check_resource_for_domains.apply_async(
+                kwargs=dict(
+                    guid=get_guid(item),
+                    content=spam_content,
+                )
+            )
+            logger.info(f'{item}, queued')
+
+
+def backfill_domain_references(date_modified, model_name, dry_run=False, batch_size=None):
+    model = apps.get_model(model_name)
+    query = reduce(
+        operator.or_,
+        (Q(**{f'{field}__regex': DOMAIN_SEARCH_REGEX}) for field in list(model.SPAM_CHECK_FIELDS))
+    ) & Q(modified__lte=date_modified)
```

Comment:
If the only exclusion criteria is `modified__lte`, having a `batch_size` could lead to the same query results repeatedly. 

Should also exclude on the existence of `DomainReference` objects, as that would indicate this object has been through a `check_resource_for_domains` already.

Reply:
At the very least, we need an `item.save` in the `spawn_tasks` function to get that update to `modified`. But, since we're acting on models that people actually care about the `modified` date on, subquerying for existing `DomainReferences` is probably a safer bet. Then we could exclude the `date_modified` check altogether?

Reply:
Wikis are the sticky wicket here, since they won't have any direct `DomainReferences`...